### PR TITLE
Use official upload release asset

### LINKED ISSUE #127: use actions/upload-release-asset instead of svenstaro
Someone should take time to change these upload actions to `actions/upload-release-asset`.

_Originally posted by @mzabaluev in https://github.com/input-output-hk/chain-wallet-libs/pull/126#discussion_r535914953_

### PR DESCRIPTION
**Pull request by enzoc4**
closes #127

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -64,13 +67,6 @@ jobs:
           override: true
           default: true
 
-      - name: Downgrade cross
```
Comment (mzabaluev) [inline .github/workflows/release.yaml:67]: This does not seem related; was this cleanup meant to be included in the PR?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -118,15 +114,26 @@ jobs:
...
       - name: Upload binaries to release
-        uses: svenstaro/upload-release-action@v1-release
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: chain-wallet-libs-${{ steps.get_version.outputs.VERSION }}-${{ matrix.config.target }}.*
+          upload_url: ${{ needs.initial_release.outputs.upload_url }}
+          asset_path: chain-wallet-libs-${{ steps.get_version.outputs.VERSION }}-${{ matrix.config.target }}.${{ steps.get_file_extension.outputs.EXT }}
```
Comment (mzabaluev) [inline .github/workflows/release.yaml:134]: It would be better if all output names used the same capitalization, like `upload_url` above. All caps looks odd in these expressions.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -170,16 +177,16 @@ jobs:
...
+          upload_url: ${{ needs.initial_release.outputs.upload_url }}
+          asset_path: chain-wallet-libs-${{ steps.get_version.outputs.VERSION }}-${{ matrix.target }}.tgz
           asset_name: chain-wallet-libs-${{ steps.get_version.outputs.VERSION }}-${{ matrix.target }}
-          tag: ${{ github.ref }}
-          file_glob: true
-          overwrite: true
+          asset_content_type: application/gzip
 
-  release_keygen:
+  release-keygen:
```
Comment (mzabaluev) [inline .github/workflows/release.yaml:189]: This renaming seems arbitrary, too, as all other step names use snake_case.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (enzoc4) [reply .github/workflows/release.yaml:67]: Well, it's kind of arbitrary, yeah. I started deleting this in { #129 #128 }. As it doesn't seem to have any effect I just deleted this one too without thinking about it.

**Review by mzabaluev — APPROVED** [review]
